synapseclient/models/wiki.py

```diff
@@ -1 +1,804 @@
+
+        # Convert markdown to gzipped file if needed
+        file_path = self._markdown_to_gzip_file(self.markdown, synapse_client)
+
+        # Upload the gzipped file to get a file handle
+        file_handle = await upload_file_handle(
+            syn=synapse_client,
+            parent_entity_id=self.owner_id,
+            path=file_path,
+        )
+
+        # delete the temp gzip file
+        os.remove(file_path)
+
+        # Set the markdown file handle ID from the upload response
+        self.markdown_file_handle_id = file_handle.get("id")
+
+        # Create the wiki page
+        wiki_data = await post_wiki(
+            owner_id=self.owner_id,
+            request=self.to_synapse_request(),
+            synapse_client=synapse_client,
+        )
+
+        if force_version and self.wiki_version is not None:
+            wiki_data = await put_wiki_version(
+                owner_id=self.owner_id,
+                wiki_id=self.id,
+                wiki_version=self.wiki_version,
+                request=wiki_data,
+                synapse_client=synapse_client,
+            )
+
+        else:
+            raise ValueError("Must provide wiki_version to force a new version.")
+
+        self.fill_from_dict(wiki_data)
+        return self
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Wiki_Page: {self.owner_id}"
+    )
+    async def get_async(
+        self,
+        *,
+        synapse_client: Optional[Synapse] = None,
+    ) -> "WikiPage":
+        """Get a wiki page from Synapse.
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                    `Synapse.allow_client_caching(False)` this will use the last created
+                    instance from the Synapse class constructor.
+
+        Returns:
+            The wiki page.
+
+        Raises:
+            ValueError: If owner_id is not provided.
+
+        """
+        if not self.owner_id:
+            raise ValueError("Must provide owner_id to get a wiki page.")
+
+        # If we have an ID, use it directly (TO SIMPLIFY)
+        elif self.id:
+            wiki_data = await get_wiki_page(
+                owner_id=self.owner_id,
+                wiki_id=self.id,
+                wiki_version=self.version_number,
+                synapse_client=synapse_client,
+            )
+        # If we only have a title, find the wiki page with matching title
+        else:
+            results = await get_wiki_header_tree(
+                owner_id=self.owner_id,
+                synapse_client=synapse_client,
+            )
+            async for result in results:
+                if result.get("title") == self.title:
+                    matching_header = result
+                    break
+                else:
+                    matching_header = None
+
+            if not matching_header:
+                raise ValueError(f"No wiki page found with title: {self.title}")
```

Comment:
Is this the right way to use an async generator? Wouldn't the correct syntax be:

```suggestion
            async for result in get_wiki_header_tree(
                owner_id=self.owner_id,
                synapse_client=synapse_client,
            ):
                if result.get("title") == self.title:
                    matching_header = result
                    break
                else:
                    matching_header = None

            if not matching_header:
                raise ValueError(f"No wiki page found with title: {self.title}")
```

(The code is not tested)

Reply:
I think you are right. I forgot to remove the `await` before `get_wiki_header_tree`. This is my test code and I didn't specify await..
```
async def main():
    results = get_wiki_header_tree(owner_id="syn3380222", limit=5)
    async for result in results:
        if result.get("title") == 'Mutations In Cis':
            print("Found match:", result['id'])
            break
        else:
            matching_header = None
        print(matching_header)

asyncio.run(main())

```